data/battle_scripts_1.s

```diff
@@ -365,6 +365,174 @@ gBattleScriptsForMoveEffects:: @ 82D86A8
 	.4byte BattleScript_EffectFairyLock
 	.4byte BattleScript_EffectAllySwitch
 	.4byte BattleScript_EffectSleepHit
+	.4byte BattleScript_EffectJawLock
+	.4byte BattleScript_EffectNoRetreat
+	.4byte BattleScript_EffectTarShot
+	.4byte BattleScript_EffectDragonDarts
+	.4byte BattleScript_EffectPoltergeist
+	.4byte BattleScript_EffectOctolock
+	.4byte BattleScript_EffectClangorousSoul
+	.4byte BattleScript_EffectBoltBeak
+
+BattleScript_EffectClangorousSoul:
+	attackcanceler
+	attackstring
+	ppreduce
+	cutonethirdhpraisestats BattleScript_ButItFailed
+	orword gHitMarker, HITMARKER_IGNORE_SUBSTITUTE
+	attackanimation
+	waitanimation
+	healthbarupdate BS_ATTACKER
+	datahpupdate BS_ATTACKER
+	call BattleScript_AllStatsUp
+	goto BattleScript_MoveEnd
+
+BattleScript_EffectOctolock:
+	attackcanceler
+	jumpifsubstituteblocks BattleScript_ButItFailedAtkStringPpReduce
+	accuracycheck BattleScript_PrintMoveMissed, ACC_CURR_MOVE
+	attackstring
+	ppreduce
+	setoctolock BS_TARGET, BattleScript_ButItFailed
+	attackanimation
+	waitanimation
+	printstring STRINGID_CANTESCAPEBECAUSEOFCURRENTMOVE
+	waitmessage 0x30
+	goto BattleScript_MoveEnd
+
+BattleScript_OctolockEndTurn::
+	playanimation BS_ATTACKER, B_ANIM_TURN_TRAP, 0
+	jumpifstat BS_TARGET, CMP_EQUAL, STAT_DEF, MIN_STAT_STAGE, BattleScript_OctolockTryLowerSpDef
```

Comment:
if the target's def is 0 but their sp. def is not, no animation will play. I recommend something like:
```
        jumpifstat BS_TARGET, CMP_GREATER_THAN, STAT_DEF, MIN_STAT_STAGE, BattleScript_OctolockLowerDef
        jumpifstat BS_TARGET, CMP_GREATER_THAN, STAT_SPDEF, MIN_STAT_STAGE, BattleScript_OctolockLowerDef
        goto BattleScript_OctolockEnd2
BattleScript_OctolockLowerDef:
        playstatchangeanimation BS_ATTACKER, BIT_SPATK | BIT_SPDEF, STAT_CHANGE_NEGATIVE
        jumpifability BS_TARGET, ABILITY_BIG_PECKS, BattleScript_OctolockTryLowerSpDef
        setstatchanger STAT_DEF, 1, TRUE
	statbuffchange MOVE_EFFECT_AFFECTS_USER | STAT_BUFF_ALLOW_PTR, BattleScript_OctolockTryLowerSpDef
	jumpifbyte CMP_EQUAL, cMULTISTRING_CHOOSER, 0x2, BattleScript_OctolockTryLowerSpDef
	printfromtable gStatUpStringIds
	waitmessage 0x40
BattleScript_OctolockTryLowerSpDef::
	setstatchanger STAT_SPDEF, 1, TRUE
	statbuffchange MOVE_EFFECT_AFFECTS_USER | STAT_BUFF_ALLOW_PTR, BattleScript_OctolockEnd2
	jumpifbyte CMP_EQUAL, cMULTISTRING_CHOOSER, 0x2, BattleScript_OctolockEnd2
	printfromtable gStatUpStringIds
	waitmessage 0x40
BattleScript_OctolockEnd2::
	end2
```

(this might need some testing)